runtime/src/sorted_storages.rs

```diff
@@ -45,21 +45,25 @@ impl<'a> SortedStorages<'a> {
                 storage.slot() // this must be unique. Will be enforced in new_with_slots
             })
             .collect::<Vec<_>>();
-        Self::new_with_slots(source, &slots)
+        Self::new_with_slots(source, &slots, None)
     }
 
     // source[i] is in slot slots[i]
     // assumptions:
     // 1. slots vector contains unique slot #s.
     // 2. slots and source are the same len
-    pub fn new_with_slots(source: &'a [SnapshotStorage], slots: &[Slot]) -> Self {
+    pub fn new_with_slots(
+        source: &'a [SnapshotStorage],
+        slots: &[Slot],
```

Comment:
Would it be cleaner to pass an `impl Iter<Item=(SnapshotStorage, Slot)>` here, i.e. pass in a zipped iterator?

Reply:
Well, first, we iterate the slots twice to find min and max. I have considered calculating those when we generate the slots vector upstream as well to simplify this code.
Second, the data itself is currently created as 2 corresponding vectors to prevent other callers who need just a vector of SnapshotStorage to not have to split out the slot values they don't need.
I think it makes things more complicated somewhere.

Reply:
Yeah, so `get_snapshot_storages()`outputs 2 vectors, but we can zip those in `calculate_accounts_hash_helper()` before passing them here, so that shouldn't affect other callers of `get_snapshot_storages`. This seems to work: https://github.com/carllin/solana/commit/bb9ac0849a93f22e5aa526316b4a1fb8beb4a4d3

Reply:
ok. I cherry-picked your change.